cmake/OpenCVDetectPython.cmake

```diff
@@ -38,82 +32,37 @@ if(NOT ${found})
     set(PYTHON_EXECUTABLE "${${executable}}")
   endif()
 
-  if(WIN32 AND NOT ${executable} AND OPENCV_PYTHON_PREFER_WIN32_REGISTRY)  # deprecated
-    # search for executable with the same bitness as resulting binaries
-    # standard FindPythonInterp always prefers executable from system path
-    # this is really important because we are using the interpreter for numpy search and for choosing the install location
-    foreach(_CURRENT_VERSION ${Python_ADDITIONAL_VERSIONS} "${preferred_version}" "${min_version}")
-      find_host_program(PYTHON_EXECUTABLE
-        NAMES python${_CURRENT_VERSION} python
-        PATHS
-          [HKEY_LOCAL_MACHINE\\\\SOFTWARE\\\\Python\\\\PythonCore\\\\${_CURRENT_VERSION}\\\\InstallPath]
-          [HKEY_CURRENT_USER\\\\SOFTWARE\\\\Python\\\\PythonCore\\\\${_CURRENT_VERSION}\\\\InstallPath]
-        NO_SYSTEM_ENVIRONMENT_PATH
-      )
-    endforeach()
-  endif()
-
   if(preferred_version)
     set(__python_package_version "${preferred_version} EXACT")
-    find_host_package(PythonInterp "${preferred_version}" EXACT)
-    if(NOT PYTHONINTERP_FOUND)
+    find_package(Python3 "${preferred_version}" EXACT COMPONENTS Interpreter)
```

Comment:
Most probably it breaks cross builds, e.g. linux x86 -> linux arm. We need to instances of Python: python interpreter for scripting (host) and python + dev files for bindings itself.